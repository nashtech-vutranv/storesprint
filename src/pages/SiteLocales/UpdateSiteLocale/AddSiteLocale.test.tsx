import {render, fireEvent, waitFor} from '@testing-library/react'
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'
import {UpdateSiteLocale} from '../'

const localeList = Promise.resolve({
  data: [
    {
      id: 'ar_AE',
      createdAt: null,
      modifiedAt: null,
      version: 0,
      name: 'Arabic - United Arab Emirates',
    },
  ],
})

jest.mock('../../../services/SiteLocaleService', () => {
  const instance = jest.fn().mockReturnValue({
    getUnassignedLocales: jest.fn().mockImplementation(() => localeList),
    addLocale: jest.fn().mockImplementation((siteLocale) => {
      if (
        siteLocale.erpId !== 'exist-erp-id' &&
        siteLocale.url !== 'http://exist-url.com'
      ) {
        return Promise.resolve({
          data: {
            id: 'id',
            createdAt: '2022-04-10T01:55:35.868052Z',
            modifiedAt: '2022-04-10T01:55:35.868052Z',
            version: 0,
            erpId: 'valid-erp-id',
            siteId: 'site-id',
            localeId: 'ar_AE',
            platformId: 'null',
            name: 'Arabic - United Arab Emirates',
            url: 'http://valid-url.com',
            status: 'ACTIVE',
          },
        })
      }
      const invalidFields = [] as any[]
      if (siteLocale.erpId === 'exist-erp-id') {
        invalidFields.push({errorMessage: 'DUPLICATE_SITE_LOCALE_ERP_ID'})
      }
      if (siteLocale.url === 'http://exist-url.com') {
        invalidFields.push({errorMessage: 'DUPLICATE_SITE_LOCALE_URL'})
      }
      return Promise.reject({
        response: {
          data: {
            errorCode: 'FIELD_INVALID',
            invalidFields: invalidFields,
          },
          status: 409,
        },
      })
    }),
  })
  return instance
})

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    siteLocaleId: undefined,
    siteId: 'site-id',
    orgId: 'org-id',
  }),
}))

test('renders Add Site Locale', async () => {
  const tree = render(
    <MemoryRouter>
      <UpdateSiteLocale />
    </MemoryRouter>
  )
  await waitFor(() => {
    expect(tree).toMatchSnapshot()
  })
  
  const saveBtn = tree.getByText(/common_confirm_save/i)
  fireEvent.click(saveBtn)
  const urlInput = tree.container.querySelector('input[name="url"]') as Element
  fireEvent.change(urlInput, {target: {value: 'invalid-url'}})
  fireEvent.blur(urlInput)
  const erpIdInput = tree.container.querySelector(
    'input[name="erpId"]'
  ) as Element
  fireEvent.change(erpIdInput, {target: {value: 'exist-erp-id'}})
  fireEvent.blur(erpIdInput)
  fireEvent.click(saveBtn)
  await waitFor(() =>
    expect(tree.getByText(/form_validate_invalid_url/i)).toBeInTheDocument()
  )
  const switchInput = tree.container.querySelector('.p-inputswitch') as Element
  fireEvent.click(switchInput)
  await waitFor(() =>
    expect(switchInput).not.toHaveClass('p-inputswitch-checked')
  )
  fireEvent.click(switchInput)
  await waitFor(() => expect(switchInput).toHaveClass('p-inputswitch-checked'))
  fireEvent.change(urlInput, {target: {value: 'http://exist-url.com'}})
  fireEvent.click(saveBtn)
  fireEvent.change(erpIdInput, {target: {value: 'valid-erp-id'}})
  fireEvent.click(saveBtn)
  fireEvent.change(urlInput, {target: {value: 'http://valid-url.com'}})
  fireEvent.click(saveBtn)
})
