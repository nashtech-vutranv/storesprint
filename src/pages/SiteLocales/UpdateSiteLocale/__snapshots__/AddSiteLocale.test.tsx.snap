// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Add Site Locale 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="breadcrumb"
      >
        <ol
          class="breadcrumb m-0"
        >
          <li
            class="breadcrumb-item"
          >
            <a
              href="/organizations"
            >
              <span>
                navigator_title_organizations
              </span>
            </a>
          </li>
          <li
            class="breadcrumb-item"
          >
            <a
              href="/organizations/org-id"
            >
              <span>
                 
                org-id
                 
              </span>
            </a>
          </li>
          <li
            class="breadcrumb-item"
          >
            <a
              href="/organizations/org-id/sites"
            >
              <span>
                 
                sites_pate_title
                 
              </span>
            </a>
          </li>
          <li
            class="breadcrumb-item"
          >
            <a
              href="/organizations/org-id/sites/site-id"
            >
              <span>
                 
                
                 
              </span>
            </a>
          </li>
          <li
            aria-current="page"
            class="breadcrumb-item active"
          >
            <span>
               
              siteLocales_page_title
               
            </span>
          </li>
        </ol>
      </nav>
      <div
        class="card-form card"
      >
        <div
          class="card-header"
        >
          <h4
            class="card-form__title"
          >
            siteLocales_add_title
          </h4>
        </div>
        <div
          class="card-body"
        >
          <div
            class="row"
          >
            <form
              autocomplete="on"
              class="form-layout"
            >
              <div
                class="container-fluid"
              >
                
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class="required"
                      for="erpId"
                    >
                      siteLocales_form_erpId
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      class="p-inputtext p-component w-full p-inputtext-sm"
                      id="erpId"
                      maxlength="16"
                      name="erpId"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-4"
                  />
                  <div
                    class="col-8"
                  />
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class="required"
                      for="localeId"
                    >
                      siteLocales_form_locale
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <div
                      class="react-select inherit-color css-b62m3t-container"
                      id="localeId"
                    >
                      <span
                        class="css-1f43avz-a11yText-A11yText"
                        id="react-select-2-live-region"
                      />
                      <span
                        aria-atomic="false"
                        aria-live="polite"
                        aria-relevant="additions text"
                        class="css-1f43avz-a11yText-A11yText"
                      />
                      <div
                        class="react-select__control css-13cymwt-control"
                      >
                        <div
                          class="react-select__value-container react-select__value-container--has-value css-1fdsijx-ValueContainer"
                        >
                          <div
                            class="react-select__single-value css-1dimb5e-singleValue"
                          />
                          <div
                            class="react-select__input-container css-qbdosj-Input"
                            data-value=""
                          >
                            <input
                              aria-autocomplete="list"
                              aria-expanded="false"
                              aria-haspopup="true"
                              aria-label="textbox"
                              autocapitalize="none"
                              autocomplete="off"
                              autocorrect="off"
                              class="react-select__input"
                              id="react-select-2-input"
                              role="combobox"
                              spellcheck="false"
                              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                              tabindex="0"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                        <div
                          class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                        >
                          <span
                            class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
                          />
                          <div
                            aria-hidden="true"
                            class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                          >
                            <svg
                              aria-hidden="true"
                              class="css-tj5bde-Svg"
                              focusable="false"
                              height="20"
                              viewBox="0 0 20 20"
                              width="20"
                            >
                              <path
                                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                      <input
                        name="localeId"
                        type="hidden"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-4"
                  />
                  <div
                    class="col-8"
                  />
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class=""
                      for="name"
                    >
                      siteLocales_form_name
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      aria-label="disabled"
                      class="p-inputtext p-component p-disabled w-full p-inputtext-sm"
                      disabled=""
                      id="name"
                      maxlength="128"
                      name="name"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-4"
                  />
                  <div
                    class="col-8"
                  />
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class="required"
                      for="url"
                    >
                      siteLocales_form_url
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      class="p-inputtext p-component w-full p-inputtext-sm"
                      id="url"
                      maxlength="255"
                      name="url"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-4"
                  />
                  <div
                    class="col-8"
                  />
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class=""
                      for="status"
                    >
                      organization_details_status_active
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <div
                      class="p-inputswitch p-component p-inputswitch-checked"
                    >
                      <div
                        class="p-hidden-accessible"
                      >
                        <input
                          aria-checked="true"
                          aria-label="checkbox"
                          checked=""
                          name="status"
                          type="checkbox"
                        />
                      </div>
                      <span
                        class="p-inputswitch-slider"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-8 offset-4"
                  >
                    <button
                      class="me-2 btn btn-danger"
                      type="button"
                    >
                      common_confirm_cancel
                    </button>
                    <button
                      class="me-2 btn btn-success"
                      disabled=""
                      type="button"
                    >
                      common_confirm_save
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="breadcrumb"
    >
      <ol
        class="breadcrumb m-0"
      >
        <li
          class="breadcrumb-item"
        >
          <a
            href="/organizations"
          >
            <span>
              navigator_title_organizations
            </span>
          </a>
        </li>
        <li
          class="breadcrumb-item"
        >
          <a
            href="/organizations/org-id"
          >
            <span>
               
              org-id
               
            </span>
          </a>
        </li>
        <li
          class="breadcrumb-item"
        >
          <a
            href="/organizations/org-id/sites"
          >
            <span>
               
              sites_pate_title
               
            </span>
          </a>
        </li>
        <li
          class="breadcrumb-item"
        >
          <a
            href="/organizations/org-id/sites/site-id"
          >
            <span>
               
              
               
            </span>
          </a>
        </li>
        <li
          aria-current="page"
          class="breadcrumb-item active"
        >
          <span>
             
            siteLocales_page_title
             
          </span>
        </li>
      </ol>
    </nav>
    <div
      class="card-form card"
    >
      <div
        class="card-header"
      >
        <h4
          class="card-form__title"
        >
          siteLocales_add_title
        </h4>
      </div>
      <div
        class="card-body"
      >
        <div
          class="row"
        >
          <form
            autocomplete="on"
            class="form-layout"
          >
            <div
              class="container-fluid"
            >
              
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class="required"
                    for="erpId"
                  >
                    siteLocales_form_erpId
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    class="p-inputtext p-component w-full p-inputtext-sm"
                    id="erpId"
                    maxlength="16"
                    name="erpId"
                    value=""
                  />
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-4"
                />
                <div
                  class="col-8"
                />
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class="required"
                    for="localeId"
                  >
                    siteLocales_form_locale
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <div
                    class="react-select inherit-color css-b62m3t-container"
                    id="localeId"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-2-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class="react-select__control css-13cymwt-control"
                    >
                      <div
                        class="react-select__value-container react-select__value-container--has-value css-1fdsijx-ValueContainer"
                      >
                        <div
                          class="react-select__single-value css-1dimb5e-singleValue"
                        />
                        <div
                          class="react-select__input-container css-qbdosj-Input"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-haspopup="true"
                            aria-label="textbox"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class="react-select__input"
                            id="react-select-2-input"
                            role="combobox"
                            spellcheck="false"
                            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-tj5bde-Svg"
                            focusable="false"
                            height="20"
                            viewBox="0 0 20 20"
                            width="20"
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="localeId"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-4"
                />
                <div
                  class="col-8"
                />
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class=""
                    for="name"
                  >
                    siteLocales_form_name
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    aria-label="disabled"
                    class="p-inputtext p-component p-disabled w-full p-inputtext-sm"
                    disabled=""
                    id="name"
                    maxlength="128"
                    name="name"
                    value=""
                  />
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-4"
                />
                <div
                  class="col-8"
                />
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class="required"
                    for="url"
                  >
                    siteLocales_form_url
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    class="p-inputtext p-component w-full p-inputtext-sm"
                    id="url"
                    maxlength="255"
                    name="url"
                    value=""
                  />
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-4"
                />
                <div
                  class="col-8"
                />
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class=""
                    for="status"
                  >
                    organization_details_status_active
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <div
                    class="p-inputswitch p-component p-inputswitch-checked"
                  >
                    <div
                      class="p-hidden-accessible"
                    >
                      <input
                        aria-checked="true"
                        aria-label="checkbox"
                        checked=""
                        name="status"
                        type="checkbox"
                      />
                    </div>
                    <span
                      class="p-inputswitch-slider"
                    />
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-8 offset-4"
                >
                  <button
                    class="me-2 btn btn-danger"
                    type="button"
                  >
                    common_confirm_cancel
                  </button>
                  <button
                    class="me-2 btn btn-success"
                    disabled=""
                    type="button"
                  >
                    common_confirm_save
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
